src/api/SearchApi.js

```diff
@@ -169,5 +169,36 @@ export default {
         return error;
       }
     );
+  },
+  getSimilar(uri) {
+    let custom = {
+      "query": {
+        "queries": [{
+          "custom-constraint-query": {
+            "constraint-name": "similar",
+            "text": uri
+          }
+        }]
+      }
+    };
+
+    return fetch('/api/search/similar', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json'
+      },
+      body: JSON.stringify(custom),
+      credentials: 'same-origin'
+    }).then(
+      response => {
+        return response.json().then(function(json) {
+          return { response: json };
+        });
+      },
+      error => {
+        return error;
+      }
+    );
```

Comment:
As suggested in the PR against grove-ml-gradle and grove-node, we should put that custom constraint inside all.xml. You can then provide the search type as param to this function, and somehow leverage the search function in this library to make the backend call..